corehq/motech/openmrs/atom_feed.py

```diff
@@ -0,0 +1,269 @@
+        )
+
+
+def get_updatepatient_caseblock(case, patient, repeater):
+    property_map = get_property_map(repeater.openmrs_config.case_config)
+
+    fields_to_update = {}
+    for prop, (jsonpath, value_map) in property_map.items():
+        matches = jsonpath.find(patient)
+        for match in matches:
+            patient_value = match.value
+            case_value = case.get_case_property(prop)
+            new_value = value_map.get(patient_value, patient_value)
+            if case_value != new_value:
+                fields_to_update[prop] = new_value
+
+    if fields_to_update:
+        case_name = patient['person']['display']
+        return CaseBlock(
+            create=False,
+            case_id=case.get_id,
+            case_name=case_name,
+            update=fields_to_update,
+        )
+
+
+def update_patient(repeater, patient_uuid, updated_at):
+    """
+    Fetch patient from OpenMRS, submit case update for all mapped case
+    properties.
+
+    .. NOTE:: OpenMRS UUID must be saved to "external_id" case property
+
+    """
+    _assert(
+        len(repeater.white_listed_case_types) == 1,
+        'Unable to update patients from OpenMRS unless a single case type is '
+        'specified. domain: "{}". repeater: "{}".'.format(
+            repeater.domain, repeater.get_id
+        )
+    )
+    case_type = repeater.white_listed_case_types[0]
+    requests = Requests(
+        repeater.domain,
+        repeater.url,
+        repeater.username,
+        repeater.password
+    )
+    patient = get_patient_by_uuid(requests, patient_uuid)
+    case, error = importer_util.lookup_case(
+        EXTERNAL_ID,
+        patient_uuid,
+        repeater.domain,
+        case_type=case_type,
+    )
+    if error == LookupErrors.NotFound:
+        owner = get_one_commcare_user_at_location(repeater.domain, repeater.location_id)
+        _assert(
+            owner,
+            'No users found at location "{}" to own patients added from '
+            'OpenMRS atom feed. domain: "{}". repeater: "{}".'.format(
+                repeater.location_id, repeater.domain, repeater.get_id
+            )
+        )
+        case_block = get_addpatient_caseblock(case_type, owner, patient, repeater)
+
+    else:
+        _assert(
+            error != LookupErrors.MultipleResults,
+            # Multiple cases matched to the same patient.
+            # Could be caused by:
+            # * The cases were given the same identifier value. It could
+            #   be user error, or case config assumed identifier was
+            #   unique but it wasn't.
+            # * PatientFinder matched badly.
+            # * Race condition where a patient was previously added to
+            #   both CommCare and OpenMRS.
+            'More than one case found matching unique OpenMRS UUID. '
+            'domain: "{}". case external_id: "{}". repeater: "{}".'.format(
+                repeater.domain, patient_uuid, repeater.get_id
+            )
+        )
```

Comment:
Is there action required for these alerts? Just wondering what their utility is.

Reply:
This particular alert needs action. (I'll check the others.) All those scenarios would be bad: 

* Identifiers are expected to be unique and if they're not then patients are probably duplicated in OpenMRS
* If PatientFinder is configured badly we could end up with duplicates, or worse, false matches where one patient's details get written over by someone else's
* Or race conditions (very unlikely) where cases end up duplicated, including with duplicate `external_id`.

I need to document what to do about this error. It might be a good idea to add a pointer to more documentation in this error message.